spec/integration/posts_spec.rb

```diff
@@ -0,0 +1,74 @@
+    let(:second_post) { Post.create(author: user, title: 'dewdew', text: 'dewdewdew') }
+    let(:comment) { Comment.create(user:, post: first_post, text: 'Nice one') }
+    let(:like) { Like.create(user:, post: first_post) }
+    before(:each) do
+      comment.save
+      like.update_like_counter
+    end
+
+    it 'shows the basic info for the user' do
+      visit user_posts_path(user_id: user.id)
+
+      # I can see the user's username.
+      expect(page).to have_content(user.name)
+      # I can see the number of posts the user has written.
+      expect(page).to have_content("Number of posts: #{user.posts.size}")
+      # I can see a post's title.
+      expect(page).to have_content(first_post.title)
+      # I can see some of the post's body.
+      expect(page).to have_content(first_post.text)
+      # I can see the first comments on a post.
+      expect(page).to have_content(comment.text)
+      # I can see how many comments a post has.
+      # I can see how many likes a post has.
+      expect(page).to have_css('p',
+                               text: "Comments: #{first_post.comments_counter}, Likes: #{first_post.likes_counter}")
+      # I can see a section for pagination if there are more posts than fit on the view.
+      expect(page).to have_button('Pagination')
+    end
+
+    it "redirects me to that post's show page When I click on a post." do
+      visit user_posts_path(user_id: user.id)
+      click_link first_post.title
+      expect(page).to have_current_path(user_post_path(user_id: user.id, id: first_post.id))
+    end
+  end
+
+  describe 'Post show page' do
+    let(:user) { User.create(name: 'Tango', photo: 'http://myphoto') }
+    let(:first_post) { Post.create(author: user, title: 'blablabla', text: 'blablabla') }
+    let(:second_post) { Post.create(author: user, title: 'dewdew', text: 'dewdewdew') }
+    let(:comment) { Comment.create(user:, post: first_post, text: 'Nice one') }
+    let(:like) { Like.create(user:, post: first_post) }
+    before(:each) do
+      comment.save
+      like.update_like_counter
+    end
+
+    it 'shows the post info' do
+      visit user_post_path(user_id: user.id, id: first_post.id)
+      # I can see the post's title.
+      expect(page).to have_content(first_post.title)
+      # I can see who wrote the post.
+      expect(page).to have_content(first_post.author.name)
+      # I can see how many comments it has.
+      expect(page).to have_css('p', text: "Comments: #{first_post.comments_counter}")
+      # I can see how many likes it has.
+      expect(page).to have_css('p', text: "Likes: #{first_post.likes_counter}")
+      # I can see the post body.
+      expect(page).to have_content(first_post.text)
+      # I can see the username of each commentor.
+      first_post.comments.each do |cm|
+        expect(page).to have_content(cm.user.name)
+        # I can see the comment each commentor left.
+        expect(page).to have_content(cm.text)
+      end
+    end
+  end
+end
```

Comment:
- [ ] It is required that this should be splitted into two separate files. One for the user post index page and the second for the user post show page. Kindly do that.
- [ ] There's one test yet to be written for this page.

### User post index page
![user-pic](https://github.com/Muhammad0602/Blog-app/assets/26597432/f2a8c7e7-4c6e-4bfd-b64a-ee532396b54e)

Reply:
The requirements ask to group the test so that they make sense, and we did so by splitting them into different `describe` blocks. But I cannot find any requirements about creating new files for them.